### ISSUE TITLE
Prisma client is randomly closing subscriptions due to timeout

### ISSUE THREAD

**Issue by pantharshit00** [prisma/prisma#3950]
Some users are reporting that prisma client is randomly disconnecting from the prisma server. This happens because we have an `inActivityTimout` of `60000` ms that disconnects the client. While this is beneficial to avoid memory leaks if the user is not using subscriptions, but this is also leading disconnects. 

So we should expose these options in the prisma client configuration object.

**Comment by divyenduz** [comment]
The [implementation of `inactivityTimeout`](https://github.com/apollographql/subscriptions-transport-ws/pull/390) in `subscription-transport-ws` considers active subscriptions and if there are active subscriptions the inactivity timeout does not `close` the connection. 

This might be a bug in `subscription-transport-ws` implementation. 

This could also be an issue with construction/usage of subscription client in Prisma client. 

To move this forward:- 

1. We can validate the `subscription-transport-ws` implementation of `inactivityTimeout` for correctness. 
2. Validate the Prisma client usage of `SubscriptionClient`, we can wrap the [`request` method on `SubscriptionClient`](https://github.com/prisma/prisma/blob/master/cli/packages/prisma-client-lib/src/Client.ts#L228) and create a new client if the [status is closed](https://github.com/apollographql/subscriptions-transport-ws/blob/master/src/client.ts#L138).

**Comment by pantharshit00** [comment]
I don't have a reliable repro right now. Maybe someone that has this problem can provide one.

cc @joshhopkins

**Comment by joshhopkins** [comment]
I have a simple listener process running on pm2. Something like this:

```
(async function listenForNewPosts() {
    try {
        const postEvent = await prisma.$subscribe
            .post({
                mutation_in: ['CREATED'],
            })
            .node()

        for await (const post of postEvent) {
            console.log(post)
                }
            }
        }
    } catch (error) {
        throw new Error(error)
    }
})()
```

In this case events may only fire every hour. Using the Prisma client out of the box the client would disconnect and not attempt to reconnect. I was able to remedy the issue by adding the following options to [SubscriptionClient](https://github.com/apollographql/subscriptions-transport-ws#subscriptionclient)

```
 timeout: 30000,
 inactivityTimeout: 0,
 reconnect: true,
```

Can see the concern for memory leaks. In this case I'm working around that by setting `max_memory_restart` within pm2.

**Comment by divyenduz** [comment]
This is fixed the the latest alpha, `1.31.0-alpha.3`, please install it via `npm install -g prisma@alpha`
